### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by OliverNeuhaeusler**
Hello. 
please  test my search favorite, comment and Ratingfunction. i put in the backend so please look if it works.
Thank your for reviewing

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -35,5 +35,6 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
-  }
+  },
+  "porxy": "http://localhost:4000"
```
Comment (miriamhaenle) [inline client/package.json]: You probably meant 
```suggestion
  "proxy": "http://localhost:4000"
```

File: client/src/App.js [hunk 2]
```diff
@@ -3,31 +3,143 @@ import { Switch, Route } from 'react-router-dom';
 import { saveToLocalStorage, loadFromLocalStorage } from './lib/localStorage';
 import styled from 'styled-components/macro';
 import Home from './pages/Home.js';
-import BurgerMenu from './BurgerMenu.js';
+import BurgerMenu from './components/BurgerMenu.js';
 import MarketCard from './pages/MarktCard.js';
 import MarketForm from './pages/MarktForm.js';
```
Comment (miriamhaenle) [inline client/src/App.js]: It is often seen as good practice to structure the imports from outside in:
First external imports (e.g. `styled`)
And then your own components, libs etc. I would also recommend that you group them (all pages together) and sort them alphabetically. Just gives a very structured impression :)

File: client/src/App.js [hunk 3]
```diff
@@ -3,31 +3,143 @@ import { Switch, Route } from 'react-router-dom';
...
+            setMarkets([...upToDateMarkets, updatedMarket]);
+          })
+          .catch((error) => console.error(error));
       }
       return market;
     });
+  }
 
-    setMarkets(updatedMarkets);
+  function addRating(rating, marketToUpdate) {
```
Comment (miriamhaenle) [inline client/src/App.js]: Your functions `addComment` and `addRating` are very similar. The only distinction being, that the one updates the ratings and the other the comments, but the rest of the code is almost identical. 
Keeping the DRY principles in mind, this calls for refactoring 😄 Maybe you find a way to improve this to have one function that handles both use cases. Not something that needs to be done in this PR but keep that as possible improvement for a future sprint.

File: client/src/App.js [hunk 4]
```diff
@@ -3,31 +3,143 @@ import { Switch, Route } from 'react-router-dom';
...
 
+  useEffect(() => {
+    fetch('http://localhost:4000/market')
+      .then((result) => result.json())
+      .then((marketFromApi) => setMarkets(marketFromApi))
+      .catch((error) => console.error(error));
+  }, []);
+
+  useEffect(() => {
+    setFilteredMarkets(markets);
```
Comment (miriamhaenle) [inline client/src/App.js]: Why the additional `useEffect` here?
Both your state update function (`setFilteredMarkets`) and your local storage function (`saveToLocalStorage`) have the `markets` as dependency array. So you could put both into one `useEffect` :)

File: client/src/App.js [hunk 5]
```diff
@@ -3,31 +3,143 @@ import { Switch, Route } from 'react-router-dom';
...
+  }, [markets]);
+
   useEffect(() => {
     saveToLocalStorage('Markets', markets);
   }, [markets]);
 
+  useEffect(() => {
```
Comment (miriamhaenle) [inline client/src/App.js:30]: Your structure is a bit of a mixture of useEffects and functions. 
I would recommend that you put all hooks on top of your component (`useState` and `useEffect`) .
Afterwards the functions can follow. 
But also consider extracting the functions from the file and importing them :)

File: client/src/App.js [hunk 6]
```diff
@@ -42,11 +154,32 @@ function App() {
...
+            {filteredMarkets &&
+              filteredMarkets.map((filteredMarket) => (
+                <MarketCard
+                  market={filteredMarket}
+                  onAddComment={addComment}
+                  onAddRating={addRating}
+                  onAddToFav={toggleFav}
+                  isFavorite={isFavorite}
+                />
+              ))}
```
Comment (miriamhaenle) [inline client/src/App.js]: Could be extracted to a page to keep the `App.js` smaller and hide the details within that component/page

File: client/src/App.js [hunk 7]
```diff
@@ -42,11 +154,32 @@ function App() {
...
+          </Route>
+          <Route path="/favorites">
+            {bookmarkedMarkets.map((market) => (
+              <MarketCard
+                market={market}
+                onAddComment={addComment}
+                onAddToFav={toggleFav}
+                isFavorite={isFavorite}
```
Comment (miriamhaenle) [inline client/src/App.js]: Same as previous comment :)

File: client/src/components/Login.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        <button block size="lg" type="submit" disabled={!validateForm()}>
+          Login
+        </button>
+      </form>
+    </LoginForm>
+  );
+}
+
+const LoginForm = styled.section`
+  display: fles;
```
Comment (miriamhaenle) [inline client/src/components/Login.js]: Haha, should not actually be a flexbox at the moment :P 
```suggestion
  display: flex;
```

File: client/src/components/Login.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+  }
+
+  return (
+    <LoginForm>
+      <form onSubmit={handleSubmit}>
+        <article size="lg" controlId="email">
```
Comment (miriamhaenle) [inline client/src/components/Login.js]: What do you use the `size` attribute for?

**Review by miriamhaenle — CHANGES_REQUESTED** [review]
Hi Oli, really great job you did here 👏  

I only had a few remarks. Your App.js is pretty huge at the moment, see what you maybe can extract from there to make it a bit more concise.

Keep up the good work :)

File: client/src/App.js (hunk 2 above)
Comment (OliverNeuhaeusler) [reply client/src/App.js]: Thnak you Miriam i changed it now

**Comment by OliverNeuhaeusler** [comment]
Thank you for reviewing my Code.  iam glad that you like it and i changed the things you said.

**Review by anitaschiller — APPROVED** [review]
Good work and nice refactoring, can be merged! :-)

**Pull request merged**